controller/src/main/java/org/jboss/as/controller/ObjectListAttributeDefinition.java

```diff
@@ -131,14 +134,16 @@ public ModelNode resolveValue(ExpressionResolver resolver, ModelNode value) thro
         ModelNode superResult = value.getType() == ModelType.LIST ? value : super.resolveValue(resolver, value);
 
         // If it's not a LIST (almost certainly UNDEFINED), then nothing more we can do
+        ModelNode result = new ModelNode();
         if (superResult.getType() != ModelType.LIST) {
             return superResult;
+        }else{
+            result.setEmptyList();
         }
 
         // Resolve each element.
         // Don't mess with the original value
         ModelNode clone = superResult == value ? value.clone() : superResult;
-        ModelNode result = new ModelNode();
```

Comment:
Nit:

Can the changes at L137-141 be removed and then here we have:

ModelNode clone = superResult == value ? value.clone() : superResult;
ModelNode result = new ModelNode();
result.setEmptyList();

Just seems a bit easier to understand with 'result' not being introduced in the middle of the logic for checking it's a LIST